libraries/AP_Common/missing/string.h

```diff
@@ -0,0 +1,15 @@
+#include_next <string.h>
+
+// Necessary for toolchains that does not provide `strndupa`, such as musl.
+#if !defined(HAVE_DECL_STRNDUPA) && !defined(strndupa)
+// The last value of the GCC extension "statement exprs" will be
+// evaluated and returned, E.g: `#define foo(n) ({ n; })` is equivalent for
+// `auto foo(auto n) { return n; }`
+#define strndupa(old_string, size)                                    \
+({                                                                    \
+    const size_t string_size = strnlen(old_string, size);             \
+    char *new_string = static_cast<char*>(alloca(string_size + 1));   \
+    new_string[string_size] = '\0';                                   \
+    static_cast<char*>(memcpy(new_string, old_string, size));         \
+})
```

Comment:
nitpick: size usually refers to the array size, including the \0. len usually refers to the length as returned by strlen/strnlen.

Reply:
Done